tests/gcp/app/app_test.go

```diff
@@ -33,62 +33,87 @@ const (
 )
 
 var (
-	address   string
-	projectID string
+	address        string
+	projectID      string
+	logadminClient *logadmin.Client
```

Comment:
Don't pass these as global variables. See my remarks about the `Retry` function.